db/migrate/20231018162600_create_audit_logs.rb

```diff
@@ -0,0 +1,13 @@
+class CreateAuditLogs < ActiveRecord::Migration[6.0]
+  def change
+    create_table :audit_logs do |t|
+      t.string :action, null: false
+      t.string :affected_resource, null: false
+      t.datetime :timestamp, null: false
+      t.string :user_ip, null: false
```

Comment:
Consider adding an index to the `timestamp` column in the `audit_logs` table to improve query performance when filtering by date range.